### ISSUE TITLE
Segmentation Fault: Pretraining Epoch 0

### ISSUE THREAD

**Issue by taha-m45160** [shrutirij/ocr-post-correction#4]
The training process is interrupted by a segmentation fault during the very first epoch as part of the pretraining process. The error encountered is as follows:

`[dynet] random seed: 1678755796`
`[dynet] allocating memory: 512MB`
`[dynet] memory allocation done.`
`[dynet] random seed: 2822143777`
`[dynet] using autobatching`
`[dynet] allocating memory: 10000MB`
`[dynet] memory allocation done.`
`train_single-source.sh: line 65: 1310309 Segmentation fault      (core dumped) python3 postcorrection/multisource_wrapper.py --dynet-mem $dynet_mem --dynet-autobatch 1 --pretrain_src1 $pretrain_src --pretrain_tgt $pretrain_tgt $params --single --vocab_folder $expt_folder/vocab --output_folder $expt_folder --model_name $pretrained_model_name --pretrain_only`

I was able to narrow down the problem to the following piece of code:
`
batch_loss.backward()
` in line 80: lm_trainer.py

The following issue may hint towards the potential problem:
[(https://github.com/clab/dynet/issues/308)]

Has anyone encountered this problem before?

**Comment by shrutirij** [comment]
You might have to allocate more memory to dynet -- how large is your training sample?

**Comment by shrutirij** [comment]
> The training sample is a little more than 0.2 million lines. I allocated memory upto 25 GB but to no avail.

Are you able to try it with a smaller dataset? Maybe just 10 lines to make sure there are no other issues with the installation?

**Comment by taha-m45160** [comment]
It didn't work with a smaller dataset either, I reinstalled dynet and it resolved the problem.
